src/libYARP_sig/include/yarp/sig/PointCloud.h

```diff
@@ -0,0 +1,597 @@
+        }
+
+        /**
+         * @brief Obtain the point given by the index.
+         * @param i, index
+         */
+        inline T& operator()(size_t i) {
+            if (i > data.size())
+            {
+                return nulldata;
+            }
+            return data[i];
+        }
+
+        template<class T1>
+        /**
+         * Assignment operator.
+         * @brief Clones the content of another image.
+         * @param alt the image to clone
+         */
+        const PointCloud<T>& operator=(const PointCloud<T1>& alt)
+        {
+            copy(alt);
+            return *this;
+        }
+
+
+        /**
+         * \brief Concatenate a point cloud to the current cloud.
+         *  \param rhs the cloud to add to the current cloud
+         * \return the new cloud as a concatenation of the current cloud and the new given cloud
+         */
+        inline PointCloud<T>&
+        operator+=(const PointCloud<T>& rhs)
+        {
+            //TODO: take the newest timestamp
+
+            yAssert(getPointType() == rhs.getPointType());
+
+            size_t nr_points = data.size();
+            data.resize (nr_points + rhs.size());
+            for (size_t i = nr_points; i < data.size (); ++i)
+            {
+                data[i] = rhs.data[i - nr_points];
+            }
+
+            header.width = data.size();
+            header.height   = 1;
+            if (rhs.isDense() && isDense())
+            {
+                header.isDense = true;
+            }
+            else
+            {
+                header.isDense = false;
+            }
+            return (*this);
+        }
+
+        /**
+         * \brief Concatenate a point cloud to another cloud.
+         * \param rhs the cloud to add to the current cloud
+         * \return the new cloud as a concatenation of the current cloud and the new given cloud
+         */
+        inline const PointCloud<T>
+        operator + (const PointCloud<T>& rhs)
+        {
+          return (PointCloud<T> (*this) += rhs);
+        }
+
+        inline void push_back (const T& pt)
+           {
+             data.push_back(pt);
+             header.width = data.size();
+             header.height = 1;
+           }
+
+        /**
+         * @brief Clear the data.
+         */
+        virtual inline void clear()
+        {
+            data.clear();
+            header.width = 0;
+            header.height = 0;
+        }
+
+        /**
+         * @brief Copy the content of an external PointCloud.
+         * @param source, pointer to the source data.
+         * @param type, enum representing the type of the source cloud.
+         * @param width, width of the source cloud.
+         * @param height, height of the source cloud.
+         * @param isDense
+         */
+        virtual void fromExternalPC(const char* source, int type, size_t width, size_t height, bool isDense = true)
+        {
+            yAssert(source);
```

Comment:
As before, are you sure that crashing is a sensible behavior (especially if not documented) for wrong parameters?

Reply:
`yarp::sig::Image*` classes are plenty of these `yAssert` :sweat_smile:
I think that the assert is necessary in these cases, but probably it is better to document it

Reply:
I see, then it is a discussion worth having on another place.